### PR TITLE
Remove blur from android #20249

### LINKED ISSUE #20249: Remove blur from android
Following the discussion here: 
https://github.com/status-im/status-mobile/issues/20066

We should remove blur from android in order to enable the new architecture.
The consensus is that we should use a solid colour, please align with design to make sure they are ok with the choice.

Coordinate with @ulisesmac @ilmotta as they have looked into it and likely they have informed opinions on the subject.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #20249

QA please look through all the screens and see if you find anything suspicious with colors or opacity of any element, thank you

@xAlisher @Francesca-G solid colors `colors/white` and `colors/neutral-80` have been used and your approval needed , thank you

| Login  | Toast | Chat |
| ------------- | ------------- | ------------- |
| image | image | image |

### CODE REVIEW

**Comment by siddarthkay** [comment]
nice you can also get rid of this then -> https://github.com/status-im/status-mobile/blob/2c6ab52a1aa651b675f296cafbf44ebb7d29d20f/react-native.config.js#L19-L23

**Comment by siddarthkay** [comment]
> you can also get rid of this then

maybe not. Android build will fail otherwise 🥲🥲🥲

**Comment by ilmotta** [comment]
Hey @flexsurfer, I did a quick check and found some problems you might be aware already (screens from Android 33):

- Incorrect blur onboarding screens (e.g. account creation) [example 1](https://github.com/status-im/status-mobile/assets/46027/6e014478-528f-40f0-8700-c65f5fb795d1), [example 2](https://github.com/status-im/status-mobile/assets/46027/63f3f57d-2ad2-47fd-8a45-838e2c8dd9b7)
- Due to slight transparency on the AC modal background, the user can see "clear text" behind the "Notifications" word. Same problem for the `Share` modal, i.e. the few % transparency only makes the UI more confusing because the text behind the modal is crisp. That's one problem I think we can't safely replace blur effects with raw transparency. In the parent issue we agreed on using solid (and opaque) color replacements if I'm not mistaken. 
image

**Comment by ilmotta** [comment]
@flexsurfer The chat header also has an issue with transparency:

**Comment by flexsurfer** [comment]
thank you @ilmotta i'll take a look, could you please elaborate about [example 2](https://github.com/status-im/status-mobile/assets/46027/63f3f57d-2ad2-47fd-8a45-838e2c8dd9b7) whats wrong with it?

**Comment by ilmotta** [comment]
> thank you @ilmotta i'll take a look, could you please elaborate about [example 2](https://github.com/status-im/status-mobile/assets/46027/63f3f57d-2ad2-47fd-8a45-838e2c8dd9b7) whats wrong with it?

On example-2 I think the text is generally less readable to my brain. I think because the background image is too colorful and noisy/busy. Maybe it's just me, but to the point that it feels wrong.

I'd say it's important to get approval from a designer in places where we're playing with transparency like in the onboarding before merging this PR @flexsurfer. If they're okay with the outcome then we're good. Just because pure transparency is tricky and can make UI worse quite easily.

**Comment by flexsurfer** [comment]
> > thank you @ilmotta i'll take a look, could you please elaborate about [example 2](https://github.com/status-im/status-mobile/assets/46027/63f3f57d-2ad2-47fd-8a45-838e2c8dd9b7) whats wrong with it?
> 
> On example-2 I think the text is generally less readable to my brain. I think because the background image is too colorful and noisy/busy. Maybe it's just me, but to the point that it feels wrong.
> 
> I'd say it's important to get approval from a designer in places where we're playing with transparency like in the onboarding before merging this PR @flexsurfer. If they're okay with the outcome then we're good. Just because pure transparency is tricky and can make UI worse quite easily.

yes, thank you, I totally agree that we need a design review before merging, and it seems like we have to adjust every case individually, just one general background option doesn't work well

at first, I was using 0.92 for example2 and it was not great so I increased it up to 0.97, but it seems like it still might not be good enough so we can go 0,98 or 0,99 so it will have almost no opacity

**Comment by flexsurfer** [comment]
so I'll probably do 0,98 by default and it should be a safe option, should work better for most cases

**Comment by ilmotta** [comment]
> > > thank you @ilmotta i'll take a look, could you please elaborate about [example 2](https://github.com/status-im/status-mobile/assets/46027/63f3f57d-2ad2-47fd-8a45-838e2c8dd9b7) whats wrong with it?
> > 
> > 
> > On example-2 I think the text is generally less readable to my brain. I think because the background image is too colorful and noisy/busy. Maybe it's just me, but to the point that it feels wrong.
> > I'd say it's important to get approval from a designer in places where we're playing with transparency like in the onboarding before merging this PR @flexsurfer. If they're okay with the outcome then we're good. Just because pure transparency is tricky and can make UI worse quite easily.
> 
> yes, thank you, I totally agree that we need a design review before merging, and it seems like we have to adjust every case individually, just one general background option doesn't work well
> 
> at first, I was using 0.92 for example2 and it was not great so I increased it up to 0.97, but it seems like it still might not be good enough so we can go 0,98 or 0,99 so it will have almost no opacity

You are right, it's very tricky. I think other CCs (at least @ulisesmac that I remember) played as well with different transparency values. That's also why we went in the direction of solid colors with no transparency in place of blur, because it's simpler to get the UI right. We can always add transparency touches in the future, but for now the simplicity of solid colors should be enough for our users.

> so I'll probably do 0,98 by default and it should be a safe option, should work better for most cases

If it works well from a UI perspective I see no problem too 👍🏼

**Comment by flexsurfer** [comment]
image
image

looks good to me

**Comment by ilmotta** [comment]
> looks good to me

Yes, that's much much better @flexsurfer

**Comment by flexsurfer** [comment]
image
yeah but still probably the biggest problem is that the text is readable and it's distracting :( so it seems like we have to go with no opacity

**Comment by ilmotta** [comment]
> yeah but still probably the biggest problem is that the text is readable and it's distracting :( so it seems like we have to go with no opacity

Yep, this is always a problem, although much less since you increased the percentage. Similar to what I shared about the AC and Share modals, even a tiny percentage of transparency can create this sort of ghost effect on the text. Fully opaque seems like a safer bet.

File: src/quo/components/blur/view.cljs
Comment (ilmotta) [inline src/quo/components/blur/view.cljs]: Can be kebab-case.

File: src/quo/components/blur/view.cljs
Comment (ilmotta) [inline src/quo/components/blur/view.cljs]: Can be private.

File: src/react_native/blur.cljs
Comment (ilmotta) [inline src/react_native/blur.cljs]: When `blur` is falsey, `view` will be nil, but this will throw an exception because nil is invalid hiccup.

File: src/quo/components/blur/view.cljs
Comment (ilmotta) [inline src/quo/components/blur/view.cljs]: Touching on performance a bit, in some cases, we can avoid the slow `merge` if all we want is to use a handful of default values. For reference, interesting discussions about merge performance https://ask.clojure.org/index.php/12780/a-faster-merge

The general idea is that we can replace `merge` with one `assoc` that takes varargs combined with one map access per default prop. Also use `or` macro instead of `get` because `get` will cause the third default argument to always be evaluated since `get` is not a macro.

Suggestion:

```clojure
(defn view-android
  [{:keys [style]} & children]
  (let [theme (quo.theme/use-theme)]
    (into [rn/view
           {:style (assoc style
                          :pointer-events :box-none
                          :background-color
                          (or (:background-color style)
                              (colors/theme-alpha colors/neutral-80 0.98 colors/white 0.98 theme)))}]
          children)))
```

**Review by ilmotta — COMMENTED** [review]
@flexsurfer I left a few comments and I haven't approved because, as the parent issue recommends, it's important to first align with designers the replacement colors/transparency. But we agreed on using solid colors, so I think it's not worth pursuing transparency as @ulisesmac already tested that before during the investigation phase.

I'm not in love with the toasts. They are transparent now and the icon overlaps with the background too much. I experimented in this branch setting everything to fully opaque and it makes toasts more legible.





Another point to consider is that if the app is in light mode, a toast generated for a light background will almost fade completely with the background of shell modals, like in the example below (this is fixed if you remove `:transparency` value from toasts `background-color`):

File: src/react_native/blur.cljs
Comment (flexsurfer) [reply src/react_native/blur.cljs]: but view won't be used if blur is nil, it will happen only on Android, and on Android view is not used, but we could remove this for now, idk how this lib will be excluded for Android

File: src/quo/components/blur/view.cljs
Comment (flexsurfer) [reply src/quo/components/blur/view.cljs]: thank you good point, this is my old habit to use merge :( happens automatically

File: src/react_native/blur.cljs
Comment (ilmotta) [reply src/react_native/blur.cljs]: It should be safe enough, just a bit dangerous because it's invalid hiccup. If you think it's okay to keep this no worries then.

File: src/quo/components/blur/view.cljs
Comment (ilmotta) [reply src/quo/components/blur/view.cljs]: Same, I keep forgetting because who doesn't like the convenience of merge ;)

**Comment by flexsurfer** [comment]
thank you @ilmotta for your valuable feedback, I addressed all your points and added conditions for every case where background-color is set, so it is used only for ios (so no transparent anymore), and set default solid colors for the background, this is a much safer option and easier to negotiate it with designers

**Comment by flexsurfer** [comment]
also fixed a few cases for reanimated blur

**Review by ilmotta — APPROVED** [review]
Thank you @flexsurfer. PR looks good 🚀

**Comment by qoqobolo** [comment]
Hi @flexsurfer, thanks for the PR!
I would suggest checking this PR from the design perspective first and then going through manual QA in case any further changes need to be made. 

@Francesca-G, could you review it and ping `@status-im/mobile-qa`  for manual testing once it gets your approval, please?
Thanks!

**Review by Francesca-G — APPROVED** [review]
Looks good from what I can see, nice job 👍

only problem I've encountered (might not be related to this PR) is the "I understand my password can't be recovered" text overlapping the edge of the text box:

Screenshot 2024-06-05 alle 16 24 06

>
adding the follow up label to address it either here or in a separate issue 🙏

@status-im/mobile-qa ready for testing 🚀

**Comment by flexsurfer** [comment]
> text overlapping the edge of the text box:

thank you, not related to this PR

**Comment by qoqobolo** [comment]
Thanks for the amazing work @flexsurfer! I've checked most of the app, and so far I've only found one place that doesn't look quite as expected, please take a look:

### ISSUE 1: Gradient of the wallet account drawer in dark theme (Android)

**Steps:**
1. Switch to the dark theme on Android
2. Open wallet
3. Create at least one additional account
4. Open any of your accounts
5. Tap the ACC icon at the top right to open the drawer

We already have a similar issue on iOS https://github.com/status-im/status-mobile/issues/19533, can it be the same, wdyt?


PR:

Screenshot 2024-06-06 at 15 06 54

Nightly:

Screenshot 2024-06-06 at 15 08 19

**Comment by flexsurfer** [comment]
thank you @qoqobolo i would create a new issue or add to https://github.com/status-im/status-mobile/issues/19533, so someone from wallet team could take a look at it

**Comment by qoqobolo** [comment]
> thank you @qoqobolo i would create a new issue or add to #19533, so someone from wallet team could take a look at it

Agreed 🤝

So, I couldn't find anything except for that drawer issue, amazing work @flexsurfer!
PR can be merged.

**Pull request merged**
